### PR TITLE
MRG: Fix sklearn

### PR DESCRIPTION
**Pull request by larsoner**
This should fix the failing tests in `master`.

### CODE REVIEW

File: mne/decoding/transformer.py [hunk 1]
```diff
@@ -170,7 +170,7 @@ def transform(self, epochs_data, y=None):
         memory usage may be large with big data.
         """
         return _sklearn_reshape_apply(self._scaler.transform, True,
-                                      epochs_data, y)
+                                      epochs_data)
```
Comment (larsoner) [inline mne/decoding/transformer.py]: @kingjr is this change okay for you? It's giving errors with `sklearn` 0.19. The `Scaler.transform` shouldn't need the labels, right?

Comment (kingjr) [inline mne/decoding/transformer.py]: Ah maybe it does because of class weight. let me look into it

Comment (larsoner) [inline mne/decoding/transformer.py]: Here is the error:

https://travis-ci.org/mne-tools/mne-python/jobs/263653811#L2857

Maybe we need to pass as a `**kwargs`, i.e. `y=y` instead of as positional `*args`?

Comment (larsoner) [inline mne/decoding/transformer.py]: (I pushed a commit with this change, too, we'll see if it works)

Comment (kingjr) [inline mne/decoding/transformer.py]: Two things
1. self.transform should not take `y`, [this is the deprecation sklearn generated](https://github.com/scikit-learn/scikit-learn/blob/f9332d56/sklearn/preprocessing/data.py#L659).
2. self.fit should have a default `y=None`

Comment (larsoner) [inline mne/decoding/transformer.py]: Okay it should be fixed, can you see if you agree?

**Comment by larsoner** [comment]
I'm going to merge to get `master` green again, @agramfort feel free to look when you eventually have time and I'll make another PR to clean up if necessary
